content/blog/pulumi-components/index.md

```diff
@@ -0,0 +1,339 @@
+
+export interface SecureBucketArgs {
+    bucketName?: pulumi.Input<string>;
+    versioning?: boolean;
+    encryption?: boolean;
+    tags?: { [key: string]: pulumi.Input<string> };
+}
+
+export class SecureBucket extends pulumi.ComponentResource {
+    public readonly bucket: aws.s3.Bucket;
+    public readonly bucketName: pulumi.Output<string>;
+
+    constructor(name: string, args: SecureBucketArgs = {}, opts?: pulumi.ComponentResourceOptions) {
+        super("my:components:SecureBucket", name, {}, opts);
+
+        // Create an S3 bucket with best practices by default
+        this.bucket = new aws.s3.Bucket(`${name}`, {
+            bucket: args.bucketName,
+            tags: {
+                ManagedBy: "Pulumi",
+                ...args.tags,
+            },
+        }, { parent: this });
+
+        // Conditionally enable versioning
+        if (args.versioning !== false) {
+            new aws.s3.BucketVersioningV2(`${name}-versioning`, {
+                bucket: this.bucket.id,
+                versioningConfiguration: {
+                    status: "Enabled",
+                },
+            }, { parent: this });
+        }
+
+        // Conditionally enable encryption
+        if (args.encryption !== false) {
+            new aws.s3.BucketServerSideEncryptionConfigurationV2(`${name}-encryption`, {
+                bucket: this.bucket.id,
+                rules: [{
+                    applyServerSideEncryptionByDefault: {
+                        sseAlgorithm: "AES256",
+                    },
+                }],
+            }, { parent: this });
+        }
+
+        this.bucketName = this.bucket.id;
+
+        this.registerOutputs({
+            bucket: this.bucket,
+            bucketName: this.bucketName,
+        });
+    }
+}
+```
+
+### Sharing Components
+
+Pulumi's new Components capability is designed to make sharing infrastructure building blocks as easy as possible, whether you're publishing them to a Git repository or referencing them from a local folder in a monorepo.
+
+#### Publishing to Git
+
+Once you've authored your component, sharing it is as simple as pushing your code to a Git repository and optionally tagging a release. From there, anyone (including you) can pull it down and use it in their own Pulumi projects.
+
+For example, after pushing your SecureBucket to GitHub and tagging it as v1.0.0, others can consume it by running:
+
+```bash
+pulumi package add github.com/myorg/secure-s3-component@v1.0.0
+```
+
+Under the hood, Pulumi:
+
+1. Fetches your code from GitHub
+2. Auto-generates a local SDK from the component's schema
+3. Makes it available to your Pulumi program in your chosen language
+
+#### Referencing Components Locally
+
+For scenarios like monorepos, rapid development iterations, or when you're working with components that don't need to be published to a repository, you can reference local source code directly:
+
+```bash
+pulumi package add /path/to/local/secure-s3-component
```

Comment:
Maybe use a relative path here? I suspect that use case is more common, and I *think* people are more likely to understand that they can also use an absolute path than the other way around :thinking: